airflow/models/taskinstance.py

```diff
@@ -1528,14 +1528,19 @@ def signal_handler(signum, frame):
         Stats.incr("ti_successes")
 
     def _run_finished_callback(
-        self, callback: TaskStateChangeCallback | None, context: Context, callback_type: str
+        self,
+        callbacks: None | TaskStateChangeCallback | list[TaskStateChangeCallback],
+        context: Context,
+        callback_type: str,
     ) -> None:
         """Run callback after task finishes"""
-        try:
-            if callback:
-                callback(context)
-        except Exception:  # pylint: disable=broad-except
-            self.log.exception(f"Error when executing {callback_type} callback")
+        if callbacks:
+            callbacks = callbacks if isinstance(callbacks, list) else [callbacks]
+            for callback in callbacks:
+                try:
+                    callback(context)
+                except Exception:  # pylint: disable=broad-except
+                    self.log.exception(f"Error when executing {callback_type} callback")
```

Comment:
Add callback name here?

Reply:
`qualname` didn't work here, had to use the old `func_name`. I'm not sure why